torchaudio/datasets/quesst14.py

```diff
@@ -0,0 +1,101 @@
+import os
+import re
+from pathlib import Path
+from typing import Tuple, Union, Optional
+
+import torch
+import torchaudio
+from torch.hub import download_url_to_file
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import extract_archive
+
+
+URL = "https://speech.fit.vutbr.cz/files/quesst14Database.tgz"
+_CHECKSUM = "4f869e06bc066bbe9c5dde31dbd3909a0870d70291110ebbb38878dcbc2fc5e4"
+_LANGUAGES = [
+    "albanian",
+    "basque",
+    "czech",
+    "nnenglish",
+    "romanian",
+    "slovak",
+]
+
+
+class QUESST14(Dataset):
+    """Create QUESST14 Dataset
+
+    Args:
+        root (str or Path): Root directory where the dataset's top level directory is found
+        language (str, optional): Language to get dataset for.
+            Options: [None, ``albanian``, ``basque``, ``czech``, `nnenglish``, ``romanian``, ``slovak``].
+            (default: ``"nnenglish"``)
+        subset (str): subset of the dataset to use. Options: ["docs", "dev", "eval"].
```

Comment:
```suggestion
        subset (str or None, optional): subset of the dataset to use. Options: ["docs", "dev", "eval"].
```

Reply:
I'm thinking of keeping subset as a required parameter, and not allowing `None` as it is a bit vague given there is the `docs` subset. It seems `docs` would generally be extracted separately and used differently than the `dev` and `eval` set, so supporting a dataset that could support docs+dev only or dev+eval only etc seems like it could be messy without having additional class variables like len(docs). I think the user can just use `ConcatDataset` if they'd like to merge multiple subsets, thoughts?

Reply:
Yeah that's also what I'm thinking. making it a required parameter is a good idea.